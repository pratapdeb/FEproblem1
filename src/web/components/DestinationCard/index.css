.column {
    padding-top: 30px;
    width: 14vw;
    display: flex;
}
.select {
    margin-top: 19px !important;
    width: 14vw;
}