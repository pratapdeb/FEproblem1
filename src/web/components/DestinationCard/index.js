import { React, Component } from "react";
import { Select, Col, Card } from 'antd';
import { isEmpty, find } from 'lodash'
import './index.css'

const { Option } = Select;
class DestinationCard extends Component {
    state = {}
    static getDerivedStateFromProps(props, state) {
        const { planets, vehicles } = props
        return {planets, vehicles}
    }
    buildOption = (options = []) => {
        return options.map(option => (<Option value={option.name} key={option.name}>{`${option.name} ${option.total_no ? '(' + option.total_no +')' :''}`}</Option>))
    }

    handlePlanetChange = (selectedPlanet) => {
        const { planets } = this.state
        const slPlanetObject = find(planets,Â ['name', selectedPlanet])
        const { distance } = slPlanetObject || []
        this.setState({distance})
        this.props.handlePlanetChange(selectedPlanet, this.props.title)
    }
    handleVehicleChange = (selectedVehicle) => {
       this.props.handleVehicleChange(selectedVehicle,this.props.title)
    }

    capacityFilter = () => {
        const { vehicles, distance } = this.state
        if(distance) { 
            const filteredVehicles =  vehicles.filter(v=> v.max_distance >= distance)
            return filteredVehicles
        }
        else return vehicles
    }
    render() {
        const { planets } = this.state
        const { title } = this.props
        const filteredVehicles  = this.capacityFilter()
        return <Col span={8} className='column'>
            <Card title={title}>
               { !isEmpty(planets) &&  <Select  className = 'select'  placeholder = 'Select Planet' onChange={this.handlePlanetChange}>
                   { this.buildOption(planets) }
                </Select>}
                { !isEmpty(filteredVehicles) && <Select  className = 'select' placeholder = 'Select Vehicle' onChange={this.handleVehicleChange}>
                    { this.buildOption(filteredVehicles) }
                </Select> }
            </Card>
        </Col>
    }
}


export default DestinationCard