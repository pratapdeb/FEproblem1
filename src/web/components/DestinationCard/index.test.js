import React from 'react';
import { Provider } from 'react-redux'
import DestinationCard from './index'
import configureStore  from 'redux-mock-store'
import renderer from 'react-test-renderer'

const mockStore = configureStore([])

describe('<DestinationCard/>', () => {

    let store 
    let component

    beforeEach(() =>  {
        store  = mockStore({})
        component = renderer.create(
            <Provider store={store}>
                <DestinationCard />
            </Provider>
        )
    })

    it('Matches the snapshot', () => {
        expect(component.toJSON()).toMatchSnapshot()
    })
})