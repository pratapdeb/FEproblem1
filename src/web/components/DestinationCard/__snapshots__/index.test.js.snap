// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DestinationCard/> Matches the snapshot 1`] = `
<div
  className="ant-col ant-col-8 column"
  style={Object {}}
>
  <div
    className="ant-card ant-card-bordered"
  >
    <div
      className="ant-card-body"
      style={Object {}}
    />
  </div>
</div>
`;
